### PR TITLE
Adds HHVM runtime error code

### PR DESCRIPTION
**Pull request by xiphiaz**
Ref: https://github.com/facebook/hhvm/blob/HHVM-3.8/hphp/runtime/base/runtime-error.h#L62

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Can you please use the actual constant.

**Comment by GrahamCampbell** [comment]
Can you do a is defined check, then use it if it is defined.

**Comment by GrahamCampbell** [comment]
We can't just use raw integers like this.

**Comment by xiphiaz** [comment]
@GrahamCampbell yes thats probably a better idea as I first tried using the constant but it wasn't defined in non-hhvm land. Checking if it is defined will work.

File: src/Application.php [hunk 1]
```diff
@@ -338,7 +338,11 @@ protected function registerErrorHandling()
      */
     protected function isFatalError($type)
     {
-        return in_array($type, [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE]);
+        $errorCodes = [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE];
+        if (defined('FATAL_ERROR')){
```
Comment (GrahamCampbell) [inline src/Application.php]: Please add a blank line before and after this if block.

Comment (xiphiaz) [inline src/Application.php]: @GrahamCampbell ok done.

File: src/Application.php [hunk 2]
```diff
@@ -338,7 +338,13 @@ protected function registerErrorHandling()
      */
     protected function isFatalError($type)
     {
-        return in_array($type, [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE]);
+        $errorCodes = [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE];
+
+        if (defined('FATAL_ERROR')){
+            $errorCodes[] = FATAL_ERROR; //add the HHVM fatal runtime error to the codes
```
Comment (GrahamCampbell) [inline src/Application.php]: not sure if we need this comment

Comment (xiphiaz) [inline src/Application.php]: @GrahamCampbell ok dropped comment

File: src/Application.php [hunk 3]
```diff
@@ -338,7 +338,13 @@ protected function registerErrorHandling()
      */
     protected function isFatalError($type)
     {
-        return in_array($type, [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE]);
+        $errorCodes = [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE];
+
+        if (defined('FATAL_ERROR')){
```
Comment (taylorotwell) [inline src/Application.php]: bad CS here.

Comment (GrahamCampbell) [inline src/Application.php]: We could just fix this after merge @taylorotwell?

**Comment by xiphiaz** [comment]
@taylorotwell @GrahamCampbell would it be easier on contributors to enforce your code style preferences with [StyleCI](https://styleci.io/) ?

**Comment by taylorotwell** [comment]
Did Graham make you say that?

**Comment by xiphiaz** [comment]
Hah no I actually use it in my own projects.

**Comment by xiphiaz** [comment]
Oh right I just checked out their repo, didn't realise @GrahamCampbell is the developer of it.

**Comment by GrahamCampbell** [comment]
> would it be easier on contributors to enforce your code style preferences with StyleCI ?

That is the plan, yes. StyleCI isn't quite ready for laravel yet though. It will be in a day or so though. :)

**Comment by xiphiaz** [comment]
:+1:

**Pull request merged**